tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs

```diff
@@ -21,21 +22,28 @@ public InfiniteTracingTestsBase(TFixture fixture, ITestOutputHelper output):base
             _fixture.TestLogger = output;
 
             _fixture.AddCommand($"InfiniteTracingTester StartAgent");
+
+            _fixture.AddCommand("RootCommands DelaySeconds 5"); // give the agent time to warm up
+
             _fixture.AddCommand($"InfiniteTracingTester Make8TSpan");
-            _fixture.AddCommand($"InfiniteTracingTester Wait");
 
+            _fixture.AddCommand("RootCommands DelaySeconds 65"); // wait for a metric harvest cycle to run
```

Comment:
In general, hardcoded delays like this waiting for a metric harvest cause tests to run longer than required. Metrics are harvested and sent during application shutdown already due to us using the `sendDataOnExit` functionality. I would suggest removing this and waiting for the log lines of interest that will have populated the metric in to the metric aggregator, then proceed with shutdown, which will harvest the metrics (without waiting 60+ seconds).

Reply:
Well... In my defense... This is just a minor refactoring of what the original test was doing before I started to refactor... ;-) 

But I definitely get your point and will rework the test to be more efficient...